### PR TITLE
Added support for confirm (message) option to submit in FormHelper

### PR DESCRIPTION
**Pull request by mvdriel**
Added support for confirm (message) option to submit (and indirect also for end) in FormHelper

### CODE REVIEW

**Comment by inoas** [comment]
While I like the feature I am not sure I like the inline javascript that for FormHelper. I find that the inline delete javascript confirm dialogs are already an issue. Could this not be solved by having a cake.js file that registers on certain css classes?

File: lib/Cake/View/Helper.php [hunk 1]
```diff
@@ -552,7 +552,7 @@ protected function _formatAttribute($key, $value, $escape = true) {
  */
 	protected function _confirm($message, $okCode, $cancelCode = '', $options = array()) {
 		$message = json_encode($message);
-		$confirm = "if (confirm({$message})) { {$okCode} } {$cancelCode}";
+		$confirm = "if (confirm({$message})) { {$okCode} } else { {$cancelCode} }";
```
Comment (markstory) [inline lib/Cake/View/Helper.php]: Won't this change the behavior of `_confirm` for user-land helpers too? The call 

```php
$this->_confirm('are you sure?', 'doYes()', 'doNo()')
```

Previously would make the following:

```js
if (confirm('are you sure?')) {
  doYes()
}
doNo();
```

But now generates:

```js
if (confirm('are you sure?')) {
  doYes()
} else {
  doNo();
}
```

That isn't going to do the same thing, which makes me think this change isn't backwards compatible.

Comment (mvdriel) [inline lib/Cake/View/Helper.php]: @markstory True, although I would argue that the current behavior is not expected. In the function documentation:

> @param string $cancelCode Code to be executed after user chose 'Cancel'

Comment (markstory) [inline lib/Cake/View/Helper.php]: Fair point. The existing behavior is what people are relying on, not the documentation. I'm reluctant to change the behavior to match the documentation this far into the life of 2.x

Comment (mvdriel) [inline lib/Cake/View/Helper.php]: @markstory Good point. I reverted the bug fix and updated the documentation.

File: lib/Cake/View/Helper.php [hunk 2]
```diff
@@ -546,7 +546,7 @@ protected function _formatAttribute($key, $value, $escape = true) {
  *
  * @param string $message Message to be displayed
  * @param string $okCode Code to be executed after user chose 'OK'
- * @param string $cancelCode Code to be executed after user chose 'Cancel'
+ * @param string $cancelCode Code to be executed after user chose 'Cancel', also executed when okCode doesn't return
```
Comment (markstory) [inline lib/Cake/View/Helper.php]: Thanks :+1:

**Comment by mirko-pagliai** [comment]
@inoas I agree with you. But currently CakePHP works well, then I'd say go ahead this PR and then make another PR to discuss whether you can apply your proposal to all cases.

File: lib/Cake/View/Helper.php (hunk 2 above)
Comment (mvdriel) [inline lib/Cake/View/Helper.php]: @markstory Do you want another PR with the bug fix for Helper::_confirm for CakePHP 2.9?

Comment (markstory) [inline lib/Cake/View/Helper.php]: Won't it still be a behavior breaking change?

Comment (mvdriel) [inline lib/Cake/View/Helper.php]: @markstory Yes, it will, but doesn't every bug fix cause a behavior breaking change? :)

Comment (markstory) [inline lib/Cake/View/Helper.php]: That is one way of looking at it :smile: My concern is that adding the `else` will create more issues as people are used to how it works now.

Comment (tersmitten) [inline lib/Cake/View/Helper.php]: It might cause problems, but it still seems like a bug to me. Why not fix (in `2.9`) it and mention it in the release notes like before (e.g. [2-7-migration-guide](http://book.cakephp.org/2.0/en/appendices/2-7-migration-guide.html))?
